### PR TITLE
policyfilter improvements

### PR DESCRIPTION
**Pull request by kkourt**
This PR includes several improvements for polcyfilter.

* fix an issue where we did not removed policies from the policyfilter state
* add e2e tests for namespaced policies and pod label filters
* list policies in bugtool
* adds some (simple for now) metrics

There were some modifications needed in the e2e tests for the second bullet. Please review patch-by-patch.

### CODE REVIEW

File: tests/e2e/checker/rpcchecker.go [hunk 1]
```diff
@@ -229,6 +229,7 @@ func (rc *RPCChecker) check(ctx context.Context, allowList, denyList []*tetragon
 		checker, ok := rc.checker.(interface{ GetRemainingChecks() []ec.EventChecker })
 		if !ok {
 			klog.ErrorS(fmt.Errorf("checker has no method GetRemainingChecks()"), "unable to dump remaining checks")
+			return
```
Comment (willfindlay) [inline tests/e2e/checker/rpcchecker.go:229]: Nice catch, thanks!

File: tests/e2e/helpers/grpc/grpc.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	connCtx, connCancel := context.WithTimeout(ctx, 1*time.Second)
+	defer connCancel()
+
+	maxTries := 10
+	for podName, grpcPort := range tetraPorts {
+		addr := fmt.Sprintf("127.0.0.1:%d", grpcPort)
+		// NB(kkourt): maybe it would make sense to cache the grpc connections in the
+		// context, but we keep things simple for now.
```
Comment (willfindlay) [inline tests/e2e/helpers/grpc/grpc.go:33]: Makes sense to me, but yeah let's do it as a follow-up.

**Review by willfindlay — APPROVED** [review]
e2e tests and e2e changes LGTM. Thanks a lot for the bug fixes btw :rocket:

File: pkg/metrics/policyfilter/policyfilter.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+	"github.com/cilium/tetragon/pkg/metrics/consts"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	PolicyFilterOpMetrics = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name:        consts.MetricNamePrefix + "policyflter_metrics",
```
Comment (tpapagian) [inline pkg/metrics/policyfilter/policyfilter.go:18]: Maybe use `_total` suffix here? (i.e. https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#suffixes)

File: cmd/tetra/tracingpolicy/tracingpolicy.go [hunk 4]
```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/cilium/tetragon/api/v1/tetragon"
 	"github.com/cilium/tetragon/cmd/tetra/common"
 	"github.com/cilium/tetragon/pkg/btf"
+	"github.com/cilium/tetragon/pkg/logger"
```
Comment (tpapagian) [inline cmd/tetra/tracingpolicy/tracingpolicy.go:26]: nit: This seems to be related to the next commit, not this one.

File: pkg/sensors/manager.go [hunk 5]
```diff
@@ -11,7 +11,6 @@ import (
 	"github.com/cilium/tetragon/api/v1/tetragon"
 	"github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
 	"github.com/cilium/tetragon/pkg/logger"
-	sttManager "github.com/cilium/tetragon/pkg/stt"
```
Comment (tpapagian) [inline pkg/sensors/manager.go:14]: I don't have the full context about what is `stt` but it seems that `pkg/stt/sttmanager.go` is not in use by any other code. Maybe we can remove `pkg/stt/sttmanager.go` as well in this commit?

**Review by tpapagian — APPROVED** [review]
LGTM, thanks!

File: pkg/sensors/manager.go (hunk 5 above)
Comment (kkourt) [reply pkg/sensors/manager.go:14]: It relates to stack traces. Idea is to add them back at some point, so I thought maybe keep the code in the tree for now. We can always revisit this.

File: cmd/tetra/tracingpolicy/tracingpolicy.go (hunk 4 above)
Comment (kkourt) [reply cmd/tetra/tracingpolicy/tracingpolicy.go:26]: thanks!

File: pkg/metrics/policyfilter/policyfilter.go (hunk 3 above)
Comment (kkourt) [reply pkg/metrics/policyfilter/policyfilter.go:18]: done, thanks!

**Pull request merged**